src/urllib3/util/util.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional, Union
+
+
+def to_bytes(
+    x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None
+) -> bytes:
+    if isinstance(x, bytes):
+        return x
+    elif not isinstance(x, str):
+        raise TypeError(f"not expecting type {type(x).__name__}")
+    elif encoding or errors:
+        return x.encode(encoding or "utf-8", errors=errors or "strict")
+    return x.encode()
+
+
+def to_str(
+    x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None
+) -> str:
+    if isinstance(x, str):
+        return x
+    elif not isinstance(x, bytes):
+        raise TypeError(f"not expecting type {type(x).__name__}")
+    elif encoding or errors:
+        return x.decode(encoding or "utf-8", errors=errors or "strict")
+    return x.decode()
```

Comment:
You can remove one branch here:

```suggestion
    return x.decode(encoding or "utf-8", errors=errors or "strict")
```